[GSoD] Refactoring the documentation files of gRPC-Gateway (#1851)
grpc-ecosystem/grpc-gateway: docs/docs/tutorials/learn_more.md

---
layout: default
title: Learn More
nav_order: 5
parent: Tutorials
---

# Learn More

## How it works

When the HTTP request arrives at the gRPC-Gateway, it parses the JSON data into a protobuf message. It then makes a normal Go gRPC client request using the parsed protobuf message. The Go gRPC client encodes the protobuf structure into the protobuf binary format and sends it to the gRPC server. The gRPC Server handles the request and returns the response in the protobuf binary format. The Go gRPC client parses it into a protobuf message and returns it to the gRPC-Gateway, which encodes the protobuf message to JSON and returns it to the original client.

## google.api.http

> Read more about `google.api.http` on [google.api.http](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto).

Reviewer: Reword
```suggestion
Read more about `google.api.http` in [the source file documentation](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto).
```

## HTTP and gRPC Transcoding

> Read more about HTTP and gRPC Transcoding on [API 127](https://google.aip.dev/127).

Reviewer: Confusingly, AIP is actually correct.
```suggestion
Read more about HTTP and gRPC Transcoding on [AIP 127](https://google.aip.dev/127).
```

> Read more about HTTP and gRPC Transcoding on [https://google.aip.dev/127](https://google.aip.dev/127).

Reviewer 2: I think this would look better as 

```
[API 127](https://google.aip.dev/127)
```
